py/desispec/scripts/qsoqn.py

```diff
@@ -201,63 +202,69 @@ def extract_redshift_info_from_RR(filename_redrock, targetid):
     # define the output
     redshift, err_redshift, chi2, coeffs = np.zeros(targetid.size), np.zeros(targetid.size), np.inf * np.ones(targetid.size), np.zeros((targetid.size, 10))
 
-    if len(param_RR['templates_filename']) == 0:
+    if len(param_RR['template_filenames']) == 0:
         msg = 'No Redrock templates provided'
         log.critical(msg)
         raise ValueError(msg)
 
     # make an independant run for each quasar templates to circumvent some problem from redrock:
     # 1] cannot give two templates in redrock as input, only one or a directory
     # 2] problem with prior and template redshift range .. --> give zero result and redrock stop
-    for template_filename in param_RR['templates_filename']:
-        filename_priors = param_RR['filename_priors']
-        filename_output_rerun_RR = param_RR['filename_output_rerun_RR']
-        filename_redrock_rerun_RR = param_RR['filename_redrock_rerun_RR']
 
-        # find redshift range of the template
+    filename_priors = param_RR['filename_priors']
+    filename_output_rerun_RR = param_RR['filename_output_rerun_RR']
+    filename_redrock_rerun_RR = param_RR['filename_redrock_rerun_RR']
+
+    # find redshift range spanned by templates
+    zmin = 100
+    zmax = -100
+    for template_filename in param_RR['template_filenames']:
         redshift_template = fitsio.FITS(template_filename)['REDSHIFTS'][:]
-        zmin, zmax = np.min(redshift_template), np.max(redshift_template)
-        sel = (z_qn >= zmin) & (z_qn <= zmax)
+        zmin = min(zmin, np.min(redshift_template))
+        zmax = max(zmax, np.max(redshift_template))
+
+    sel = (z_qn >= zmin) & (z_qn <= zmax)
 
-        # In case where all the objects have priors outside the redshift template range
-        # Skip the template to avoid any undesired errors
-        if sel.sum() != 0:
-            write_prior_for_RR(targetid[sel], z_qn[sel], z_prior[sel], filename_priors)
+    # In case where all the objects have priors outside the redshift template range
+    # Skip the template to avoid any undesired errors
+    if sel.sum() != 0:
```

Comment:
Agreed, but I already have it implemented in another branch for another PR, so I'm not going to update it here yet since that would cause merge conflicts.  i.e. punting to another forthcoming PR.